polkadot/primitives/src/v7/executor_params.rs

```diff
@@ -175,6 +204,24 @@ impl ExecutorParams {
 		ExecutorParamsHash(BlakeTwo256::hash(&self.encode()))
 	}
 
+	/// Returns hash of preparation-related executor parameters
+	pub fn prep_hash(&self) -> ExecutorParamsPrepHash {
+		use ExecutorParam::*;
+
+		let mut enc = b"prep".to_vec();
+		for param in &self.0 {
+			if !matches!(
+				param,
+				MaxMemoryPages(..) |
+					StackNativeMax(..) | PrecheckingMaxMemory(..) |
+					PvfExecTimeout(..)
+			) {
+				enc.extend(param.encode())
+			}
+		}
```

Comment:
WDYT if we map every param at first, so if a new variant was added we can't miss it because we catch an error here?

```suggestion
		let relevant_params: Vec<ExecutorParam> = self
			.0
			.iter()
			.map(|param| match param {
				MaxMemoryPages(..) => None,
				StackLogicalMax(v) => Some(StackLogicalMax(*v)),
				StackNativeMax(..) => None,
				PrecheckingMaxMemory(..) => None,
				PvfPrepTimeout(v1, v2) => Some(PvfPrepTimeout(*v1, *v2)),
				PvfExecTimeout(..) => None,
				WasmExtBulkMemory => Some(WasmExtBulkMemory),
			})
			.flatten()
			.collect();
		for param in relevant_params {
			enc.extend(param.encode());
		}
```

Reply:
Nice idea, thank you!